apps/sms/views/shared/js/startup.js

```diff
@@ -105,8 +104,13 @@ var Startup = exports.Startup = {
       // dispatch contentInteractive when all the modules initialized
       SilentSms.init();
 
-      ActivityHandler.init();
-      SystemMessageHandler.init();
+      // When app is run as inline activity we don't want to handle any other
+      // system messages other than "activity" system message.
+      if (navigator.mozHasPendingMessage('activity')) {
```

Comment:
I just added mozHasPendingMessage('notification') in the navigation.init because of Bug 1192263. Do you think we should move this to startup and handling the status in startup instead of navigation?